pkg/v1/cli/command/core/doc.go

```diff
@@ -10,46 +10,62 @@ import (
 	"os"
 	"text/template"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/spf13/cobra/doc"
 
 	cliv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
 	cli "github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli"
 	coreTemplates "github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/command/core/templates"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/pluginmanager"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/config"
 )
 
 // DefaultDocsDir is the base docs directory
 const DefaultDocsDir = "docs/cli/commands"
+const ErrorDocsOutputFolderNotExists = "error reading docs output folder '%v', make sure folder exists or provide docs output folder as input value to '--docs-dir' argument"
 
 var (
 	docsDir string
 )
 
 func init() {
-	genAllDocsCmd.Flags().StringVarP(&docsDir, "docs-dir", "d", DefaultDocsDir, "destination for docss output")
+	genAllDocsCmd.Flags().StringVarP(&docsDir, "docs-dir", "d", DefaultDocsDir, "destination for docs output")
 }
 
 var genAllDocsCmd = &cobra.Command{
 	Use:    "generate-all-docs",
 	Short:  "Generate Cobra CLI docs for all plugins installed",
 	Hidden: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
+
+		if docsDir == "" {
+			docsDir = DefaultDocsDir
+		}
+		if _, err := os.Stat(docsDir); err != nil {
+			return errors.Wrap(err, fmt.Sprintf(ErrorDocsOutputFolderNotExists, docsDir))
+		}
```

Comment:
I suggest we also check that `docsDir` is a directory.
Currently:
```
$ touch tmpDoc. # Creates a file, not a directory
$ tanzu generate-all-docs --docs-dir tmpDoc
✖  error generate core tanzu cmd markdown "error opening tanzu.md \"open tmpDoc/tanzu.md: not a directory\""
```

Reply:
sure, i will update to check input path is a directory or not.